plugins/modules/ndo_l3out_interface_routing_policy.py

```diff
@@ -0,0 +1,724 @@
+                        "bfdMultiHopPol",
+                        "detectionMultiplier",
+                    )
+
+                    append_update_ops_data(
+                        ops,
+                        proposed_payload,
+                        interface_routing_policy_path,
+                        bfd_multi_hop_settings.get("min_receive_interval"),
+                        "bfdMultiHopPol",
+                        "minRxInterval",
+                    )
+
+                    append_update_ops_data(
+                        ops,
+                        proposed_payload,
+                        interface_routing_policy_path,
+                        bfd_multi_hop_settings.get("min_transmit_interval"),
+                        "bfdMultiHopPol",
+                        "minTxInterval",
+                    )
+
+            # BFD Settings
+            if bfd_settings is not None:
+                if bfd_settings.get("state") == "disabled" and proposed_payload.get("bfdPol"):
+                    append_update_ops_data(ops, proposed_payload, interface_routing_policy_path, None, "bfdPol", op="remove")
+
+                elif bfd_settings.get("state") != "disabled":
+                    if not proposed_payload.get("bfdPol"):
+                        append_update_ops_data(ops, proposed_payload, interface_routing_policy_path, dict(), "bfdPol")
+
+                    append_update_ops_data(ops, proposed_payload, interface_routing_policy_path, bfd_settings.get("admin_state"), "bfdPol", "adminState")
+
+                    append_update_ops_data(
+                        ops, proposed_payload, interface_routing_policy_path, bfd_settings.get("detection_multiplier"), "bfdPol", "detectionMultiplier"
+                    )
+
+                    append_update_ops_data(
+                        ops, proposed_payload, interface_routing_policy_path, bfd_settings.get("min_receive_interval"), "bfdPol", "minRxInterval"
+                    )
+
+                    append_update_ops_data(
+                        ops, proposed_payload, interface_routing_policy_path, bfd_settings.get("min_transmit_interval"), "bfdPol", "minTxInterval"
+                    )
+
+                    append_update_ops_data(
+                        ops, proposed_payload, interface_routing_policy_path, bfd_settings.get("echo_receive_interval"), "bfdPol", "echoRxInterval"
+                    )
+
+                    append_update_ops_data(
+                        ops, proposed_payload, interface_routing_policy_path, bfd_settings.get("echo_admin_state"), "bfdPol", "echoAdminState"
+                    )
+
+                    append_update_ops_data(
+                        ops,
+                        proposed_payload,
+                        interface_routing_policy_path,
+                        ENABLED_DISABLED_BOOLEAN_MAP.get(bfd_settings.get("interface_control")),
+                        "bfdPol",
+                        "ifControl",
+                    )
```

Comment:
This is my issue with the function `append_update_ops_data`. It does simplify the code but not quite enough in this case. we should write a wrapper function using  `append_update_ops_data` that should be able to do something like this:
```suggestion
                    update_data_array= [
                        [bfd_settings.get("admin_state"), ("bfdPol", "adminState")],
                        [bfd_settings.get("detection_multiplier"), ("bfdPol", "detectionMultiplier")],
                        [bfd_settings.get("min_receive_interval"), ("bfdPol", "minRxInterval")],
                        [bfd_settings.get("min_transmit_interval"), ("bfdPol", "minTxInterval")],
                        [bfd_settings.get("echo_receive_interval"), ("bfdPol", "echoRxInterval")],
                        [bfd_settings.get("echo_admin_state"), ("bfdPol", "echoAdminState")],
                        [ENABLED_DISABLED_BOOLEAN_MAP.get(bfd_settings.get("interface_control")) ("bfdPol", "ifControl")],
                    ]
                    loop_append_update_ops_data(ops, proposed_payload, interface_routing_policy_path, update_data_list)
```
with `loop_append_update_ops_data` being this new function looping over all the small sets and applying for each of them `append_update_ops_data` in `update_data_array`. 
This is a vague idea of how it could be done (like `update_data_array` could definitely have a different form) but the idea is to simplify a bit more the code if we want to go down that road.